xcodeproj/internal/templates/bazel_build.sh

```
@@ -19,8 +19,7 @@
 # - output_groups
 # - (optional) target_ids
 
-output_groups_flag="--output_groups=$(IFS=, ; echo "${output_groups[*]}")"
-readonly output_groups_flag
+readonly output_groups_flag="--output_groups='$(IFS=, ; echo "${output_groups[*]}")'"
```

Can't have readonly be on the same line as a command, or it will mask an error.

---

I'm surprised this is needed? `output_groups_flag` itself is a single string, and bazel should parse it as such.

---

fixed

---

> I'm surprised this is needed? `output_groups_flag` itself is a single string, and bazel should parse it as such.

The problem I got: ERROR: Skipping 'ios_sim_arm64-dbg-ios-sim_arm64-min13.0-applebin_ios-ST-04a1eff8420c,bi': no such target '//:ios_sim_arm64-dbg-ios-sim_arm64-min13.0-applebin_ios-ST-04a1eff8420c,bi': target 'ios_sim_arm64-dbg-ios-sim_arm64-min13.0-applebin_ios-ST-04a1eff8420c,bi' not declared in package '' defined by /Users/tttttt/Projects/app/ios/BUILD

for
/Users/tttttt/Projects/app/ios/App/App.xcodeproj/rules_xcodeproj/bazel/process_bazel_build_log.py env -i BUILD_WORKSPACE_DIRECTORY=/Users/tttttt/Projects/app/ios DEVELOPER_DIR=/nix/store/i7m8av0nxj79z857bnjwcp0cvfry4bp5-Xcode.app/Contents/Developer HOME=/Users/tttttt SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.tX9pbYjTOw/Listeners TERM=xterm-256color USER=tttttt LANG=en_US.UTF-8 PATH=/bin:/usr/bin /nix/store/wpwwfkqwh65ahd1vpbl95r0j8w2rf6pl-bazel/bin/bazel --host_jvm_args=-Xdock:name=/nix/store/i7m8av0nxj79z857bnjwcp0cvfry4bp5-Xcode.app/Contents/Developer --noworkspace_rc --bazelrc=/Users/tttttt/Projects/app/ios/App/App.xcodeproj/rules_xcodeproj/bazel/xcodeproj.bazelrc --bazelrc=.bazelrc --output_base /var/tmp/_bazel_tttttt/13aefa78eb68718c004f8ce3166ad16e/rules_xcodeproj.noindex/build_output_base build --xcode_version=16C5032a --repo_env=DEVELOPER_DIR=/nix/store/i7m8av0nxj79z857bnjwcp0cvfry4bp5-Xcode.app/Contents/Developer --repo_env=USE_CLANG_CL=16C5032a --repo_env=XCODE_VERSION=16C5032a --bes_upload_mode=NOWAIT_FOR_UPLOAD_COMPLETE --nokeep_going --config=_rules_xcodeproj_build --color=yes --output_groups='index_import,target_ids_list,bc @@//App:App ios_sim_arm64-dbg-ios-sim_arm64-min13.0-applebin_ios-ST-04a1eff8420c,bp @@//App:App ios_sim_arm64-dbg-ios-sim_arm64-min13.0-applebin_ios-ST-04a1eff8420c,bi @@//App:App ios_sim_arm64-dbg-ios-sim_arm64-min13.0-applebin_ios-ST-04a1eff8420c' @@rules_xcodeproj++internal+rules_xcodeproj_generated//generator/App/xcodeproj:xcodeproj --build_metadata=PATTERN=@@//App:App

---

with bazel=8.1.1 and rules_xcodeproj=2.11.2